apple/RNCWebView.m

```diff
@@ -185,14 +187,93 @@ - (instancetype)initWithFrame:(CGRect)frame
 
   }
 #endif // !TARGET_OS_OSX
+
+  UILongPressGestureRecognizer *longPress = [[UILongPressGestureRecognizer alloc] initWithTarget:self action:@selector(startLongPress:)];
+  longPress.delegate = self;
+
+  longPress.minimumPressDuration = 0.4f;
+  longPress.numberOfTouchesRequired = 1;
+  longPress.cancelsTouchesInView = YES;
+  [self addGestureRecognizer:longPress];
   return self;
 }
 
+- (BOOL)gestureRecognizer:(UIGestureRecognizer *)gestureRecognizer shouldRecognizeSimultaneouslyWithGestureRecognizer:(UIGestureRecognizer *)otherGestureRecognizer {
+    if ([otherGestureRecognizer isKindOfClass:[UILongPressGestureRecognizer class]]) {
+        //Only when the gesture is a long press gesture, the feedback will be blocked.
+        return YES;
+    }else{
+        return NO;
+    }
+}
+
+- (void)startLongPress:(UILongPressGestureRecognizer *)pressSender
+{
+    if(pressSender.state == UIGestureRecognizerStateBegan){
+        UIMenuController *menuController = [UIMenuController sharedMenuController];
+        NSMutableArray *menuControllerItems = [NSMutableArray arrayWithCapacity:self.menuItems.count];
+        
+        for(NSString *menuItemName in self.menuItems) {
+            NSString *sel = [NSString stringWithFormat:@"%@%@", CUSTOM_SELECTOR, menuItemName];
+            UIMenuItem *item = [[UIMenuItem alloc] initWithTitle: menuItemName
+                                                          action: NSSelectorFromString(sel)];
+            
+            [menuControllerItems addObject: item];
+        }
+  
+        menuController.menuItems = menuControllerItems;
+        [menuController setTargetRect:self.bounds inView:self];
+        [menuController setMenuVisible:YES animated:YES];
+    }
+}
+
 - (void)dealloc
 {
   [[NSNotificationCenter defaultCenter] removeObserver:self];
 }
 
+- (void)tappedMenuItem:(NSString *)eventType
+{
+    self.onSelection(@{
+        @"customMenuKey": eventType
+    });
+}
+
+- (NSMethodSignature *)methodSignatureForSelector:(SEL)sel
+{
+    if ([super methodSignatureForSelector:sel]) {
+        return [super methodSignatureForSelector:sel];
+    }
```

Comment:
Could this store `[super methodSignatureForSelector:sel]` in a local variable and return that rather than invoking the method twice?